src/dune_engine/scheduler.ml

```diff
@@ -1012,33 +1045,42 @@ end = struct
       let invalidation =
         (handle_invalidation_events events : Memo.Invalidation.t)
       in
-      (if not (Memo.Invalidation.is_empty invalidation) then (
-        match t.status with
-        | Restarting_build prev_invalidation ->
-          t.status <-
-            Restarting_build
-              (Memo.Invalidation.combine prev_invalidation invalidation)
-        | Standing_by prev ->
-          t.status <-
-            Standing_by
-              { prev with
-                invalidation =
-                  Memo.Invalidation.combine prev.invalidation invalidation
-              }
-        | Building ->
-          t.handler t.config Build_interrupted;
-          t.status <- Restarting_build invalidation;
-          Process_watcher.killall t.process_watcher Sys.sigkill
-      ) else
-        match t.status with
-        | Standing_by prev ->
-          t.status <- Standing_by { prev with saw_insignificant_changes = true }
-        | _ -> ());
-      match t.wait_for_build_input_change with
-      | None -> iter t
+      let fills =
+        if not (Memo.Invalidation.is_empty invalidation) then (
```

Comment:
Maybe we should swap the branches and get rid of the `not`, for clarity.

Reply:
Sounds good